Networkmanager: rename service to match upstream

Mic92 opened the pull request:
###### Motivation for this change

Compatibility with other distributions/software and expectation
of users coming from other systems should have higher priority over consistency.
In particular this fixes #51375, where the NetworkManager-wait-online.service
broke as a result of this.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Mic92 committed:
network-manager: rename systemd service back to match upstream

Compatibility with other distributions/software and expectation
of users coming from other systems should have higher priority over consistency.
In particular this fixes #51375, where the NetworkManager-wait-online.service
broke as a result of this.

Mic92 committed:
Revert "modemmanager: rename ModemManager.service to modem-manager.service"

This reverts commit c6882b2cf9fa13579f64d5c33cbc071fa5d5e884.

Mic92 commented:
There is one blocker though. When doing the upgrade it did not start NetworkManager again for automatically. Does anybody know why?

jtojnar commented:
Try 

```patch
--- a/pkgs/tools/networking/modem-manager/default.nix
+++ b/pkgs/tools/networking/modem-manager/default.nix
@@ -27,30 +27,16 @@
     "--with-polkit"
     "--with-udev-base-dir=$(out)/lib/udev"
     "--with-systemdsystemunitdir=$(out)/etc/systemd/system"
+    "--with-dbus-sys-dir=$(out)/etc/dbus-1/system.d"
     "--sysconfdir=/etc"
     "--localstatedir=/var"
     "--with-suspend-resume=systemd"
   ];
 
-  installFlags = [ "DESTDIR=\${out}" ];
-
-  preInstall = ''
-    mkdir -p $out/etc/systemd/system
-  '';
-
   postInstall = ''
-    # rename to modem-manager to be in style
-    mv $out/$out/etc/systemd/system/ModemManager.service $out/etc/systemd/system/modem-manager.service
-    rm -rf $out/$out/etc
-    mv $out/$out/* $out
-    DIR=$out/$out
-    while rmdir $DIR 2>/dev/null; do
-      DIR="$(dirname "$DIR")"
-    done
-
     # systemd in NixOS doesn't use `systemctl enable`, so we need to establish
     # aliases ourselves.
-    ln -s $out/etc/systemd/system/modem-manager.service \
+    ln -s $out/etc/systemd/system/ModemManager.service \
       $out/etc/systemd/system/dbus-org.freedesktop.ModemManager1.service
   '';
 
```

Mic92 commented:
Applied. This probably fixes also some references in modemmanager.

jtojnar commented:
Would keeping the legacy service name fix the transition issue? `ln -s $out/etc/systemd/system/NetworkManager.service  $out/etc/systemd/system/network-manager.service`

Anyway, even if this does not work, I would just merge this. It is not that bad.

Legogris commented:
@jtojnar 
> Would keeping the legacy service name fix the transition issue? ln -s $out/etc/systemd/system/NetworkManager.service $out/etc/systemd/system/network-manager.service

That sounds prudent to me, considering people might have started depending on the naming we are now reverting.

jtojnar commented:
> There is one blocker though. When doing the upgrade it did not start NetworkManager again for automatically. Does anybody know why?

I would say this is reasonable price to pay. User can always restart the service manually.

flokli committed:
network-manager: add legacy service name as fallback

This prevents NetworkManager from not starting back up while switching
configurations.

This can be removed once 19.10 is released.

flokli commented:
I gave @jtojnar's suggestion a try - adding the following patch:

```diff
From a201d69a825668570fbe0dc1837c229c6e70319a Mon Sep 17 00:00:00 2001
From: Florian Klink <flokli@flokli.de>
Date: Tue, 19 Mar 2019 01:54:27 +0100
Subject: [PATCH] network-manager: add legacy service name to prevent
 NetworkManager from not starting back up

This can be removed once 19.10 is released.
---
 pkgs/tools/networking/network-manager/default.nix | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/pkgs/tools/networking/network-manager/default.nix b/pkgs/tools/networking/network-manager/default.nix
index 3df029310fe..9d1ff4bb8c9 100644
--- a/pkgs/tools/networking/network-manager/default.nix
+++ b/pkgs/tools/networking/network-manager/default.nix
@@ -92,6 +92,11 @@ in stdenv.mkDerivation rec {
     # aliases ourselves.
     ln -s $out/etc/systemd/system/NetworkManager-dispatcher.service $out/etc/systemd/system/dbus-org.freedesktop.nm-dispatcher.service
     ln -s $out/etc/systemd/system/NetworkManager.service $out/etc/systemd/system/dbus-org.freedesktop.NetworkManager.service
+
+    # Add the legacy service name from before #51382 to prevent NetworkManager
+    # from not starting back up:
+    # TODO: remove this once 19.10 is released
+    ln -s $out/etc/systemd/system/NetworkManager.service $out/etc/systemd/system/network-manager.service
   '';
 
   passthru = {
-- 
2.19.2

```

And I was able to switch to the new configuration with `NetworkManager` restarting properly:

```
stopping the following units: accounts-daemon.service, bolt.service, fwupd.service, home-manager-flokli.service, network-manager.service, systemd-udevd-control.socket, systemd-udevd-kernel.socket, systemd-udevd.service, tlp.service
NOT restarting the following changed units: systemd-fsck@dev-disk-by\x2duuid-XXXX\x2dXXXX.service
activating the configuration...
setting up /etc...
reloading user units for flokli...
setting up tmpfiles
reloading the following units: dbus.service
restarting the following units: polkit.service
starting the following units: accounts-daemon.service, bolt.service, fwupd.service, home-manager-flokli.service, network-manager.service, systemd-udevd-control.socket, systemd-udevd-kernel.socket, tlp.service
the following new units were started: NetworkManager-dispatcher.service, NetworkManager.service, systemd-hostnamed.service
```

flokli commented:
rebased on staging, fixed merge commits and applied the patch from https://github.com/NixOS/nixpkgs/pull/51382#issuecomment-474157104 (with a slight change in the commit message)

jtojnar reviewed (approved):
LGTM

worldofpeace commented on pkgs/tools/networking/modem-manager/default.nix:
```diff
@@ -33,12 +33,9 @@ stdenv.mkDerivation rec {
     # systemd in NixOS doesn't use `systemctl enable`, so we need to establish
     # aliases ourselves.
-    ln -s $out/etc/systemd/system/modem-manager.service \
+    ln --relative -s $out/etc/systemd/system/ModemManager.service \
```
This didn't have any effect for me when I tested it in https://github.com/NixOS/nixpkgs/pull/57689

flokli replied:
@worldofpeace For context, idea was was the `--relative` would fix https://github.com/NixOS/nixpkgs/issues/41055.

I dropped this patch from the PR, we can follow up on that later.

worldofpeace replied:
Yep, I did similar for that pr trying to fix https://github.com/NixOS/nixpkgs/issues/41055
Probably out of scope for this too.

worldofpeace reviewed (approved):
Looks good, we should get this in.
Think it has been sufficiently tested.

flokli commented:
Well, then I'll be the one pushing the button :-)

JohnAZoidberg commented on pkgs/tools/networking/modem-manager/default.nix:
```diff
@@ -27,30 +27,16 @@ stdenv.mkDerivation rec {
     # systemd in NixOS doesn't use `systemctl enable`, so we need to establish
     # aliases ourselves.
-    ln -s $out/etc/systemd/system/modem-manager.service \
+    ln -s $out/etc/systemd/system/ModemManager.service \
```
Why resolved without comment?
Does it fix the issue?

flokli replied:
I'm not sure - can you check if the suggested change fixes #41055, and if it does, open a PR?
